tests/torch/fx/test_model_transformer.py

```diff
@@ -124,3 +131,83 @@ def test_output_insertion_transformation(tuple_output, target_point):
     check_graph(
         nncf_graph, f"output_insertion_{_target_point_to_str(target_point)}_ref.dot", TRANSFORMED_GRAPH_DIR_NAME
     )
+
+
+def count_constants(model) -> int:
+    num_constant_nodes = 0
+    for node in model.graph.nodes:
+        if node.op == "get_attr":
+            num_constant_nodes += 1
+    return num_constant_nodes
+
+
+def count_nodes_with_shared_constants(model: torch.fx.GraphModule, nncf_graph: NNCFGraph) -> int:
+    """
+    Gets the number of nodes which use a shared constant.
+    eg:
+          const
+          /   \
+    node1     node2
+
+    returns 2 (node1, and node2)
+    """
+    num_nodes_with_constant_nodes = 0
+    model_graph: torch.fx.Graph = model.graph
+    for node in model_graph.nodes:
+        nncf_node = nncf_graph.get_node_by_name(node.name)
+        num_consumer_nodes = len(nncf_graph.get_next_nodes(nncf_node))
+        if node.op == "get_attr" and num_consumer_nodes > 1:
+            assert nncf_node.is_shared()
+            num_nodes_with_constant_nodes += num_consumer_nodes
+    return num_nodes_with_constant_nodes
+
+
+def test_create_shared_constant_transformation():
+    model = MultiBranchesConnectedModel()
+    ex_inputs = torch.ones((1, 3, 3, 3))
+    captured_model = _capture_model(model, ex_inputs)
+    assert count_constants(captured_model) == 9
+    shared_constants_unification_transformation(captured_model)
+    nncf_graph = NNCFGraphFactory.create(captured_model)
+    assert count_nodes_with_shared_constants(captured_model, nncf_graph) == 3
+    assert count_constants(captured_model) == 7
+
+
+def get_shared_constant_nodes(nncf_graph: NNCFGraph):
+    """
+    Gets a dict of constant nodes as key and consumer nodes as values which are shared in the model.
+    eg:
+          const
+          /   \
+    node1     node2
+
+    returns ({const:[node1, node2]}) 
+    """
+    shared_const_node_consumer_node = {}
+    for node in nncf_graph.get_all_nodes():
+        consumer_nodes = nncf_graph.get_next_nodes(node)
+        if node.metatype in CONST_NOOP_METATYPES and len(consumer_nodes) > 1:
+            shared_const_node_consumer_node[node] = consumer_nodes
+    return shared_const_node_consumer_node
+
+
+def test_update_constant():
+    model = MultiBranchesConnectedModel()
+    ex_inputs = torch.ones((1, 3, 3, 3))
+    captured_model = _capture_model(model, ex_inputs)
+
+    shared_constants_unification_transformation(captured_model)
```

Comment:
Sounds good, I will do that!

Reply:
I have done this by calling the `count_nodes_with_shared_constants()` function before the transformation since it has an assertion for the `is_shared` attribute.

Reply:
Done

Reply:
I don't see it, please check my comment and use pytest.mark.parametrize